plugins/qcheck-stm/src/stm_of_ir.ml

```diff
@@ -143,6 +143,67 @@ let ty_show_of_core_type inst ty =
 
 let exp_of_ident id = pexp_ident (lident (str_of_ident id))
 
+(** This function should be specialized `flatten_arrow (Cfg.is_sut config)`
+    [when_sut] returns an [option] in order to notify the function whether we
+    keep the value or not *)
+let flatten_arrow is_sut when_sut when_not core_type =
+  let open Ppxlib in
+  let rec aux ty =
+    match ty.ptyp_desc with
+    | Ptyp_arrow (_, l, r) when is_sut l -> (
+        match when_sut l with None -> aux r | Some x -> x :: aux r)
+    | Ptyp_arrow (_, l, r) -> when_not l :: aux r
+    | _ -> []
+  in
+  aux core_type
+
+let gen_cmd config value =
+  (* for now we reuse the function for [ty_show], but we should at least look
+     at preconditions on [int] arguments to handle ranges *)
```

Comment:
That would be nice indeed. Would we go for a special case for `int` or is there some more generic mechanism that could be put in place there?